src/hevm/src/EVM/Solidity.hs

```diff
@@ -359,15 +359,15 @@ solidity' src = withSystemTempFile "hevm.sol" $ \path handle -> do
 -- as the codehash matches otherwise, we don't care if there is some
 -- difference there.
 stripBytecodeMetadata :: ByteString -> ByteString
-stripBytecodeMetadata bc = if BS.length cl /= 2
-                           then bc
-                           else if BS.length h >= cl' && (isRight . deserialiseFromBytes decodeTerm $ fromStrict cbor)
-                                then bc'
-                                else bc
-  where l = BS.length bc
-        (h, cl) = BS.splitAt (l - 2) bc
-        cl' = fromIntegral . runGet getWord16be . fromStrict $ cl
-        (bc', cbor) = BS.splitAt (BS.length h - cl') h
+stripBytecodeMetadata bc
+  | BS.length cl /= 2 = bc
+  | BS.length h >= cl' && (isRight . deserialiseFromBytes decodeTerm $ fromStrict cbor) = bc'
+  | otherwise = bc
```

Comment:
```suggestion
stripBytecodeMetadata bc | BS.length cl /= 2 = bc
                                         | BS.length h >= cl' && (isRight . deserialiseFromBytes decodeTerm $ fromStrict cbor) = bc'
                                         | otherwise = bc
```